src/rulesets/LinearAlgebra/dense.jl

```diff
@@ -394,3 +394,51 @@ function rrule(
     end
     return Ω, lyap_pullback
 end
+
+#####
+##### `kron`
+#####
+
+function frule((_, Δx, Δy), ::typeof(kron), x::AbstractVecOrMat{<:Number}, y::AbstractVecOrMat{<:Number})
+    return kron(x, y), kron(Δx, y) + kron(x, Δy)
+end
+
+function rrule(::typeof(kron), x::AbstractVector{<:Number}, y::AbstractVector{<:Number})
+    function kron_pullback(z̄)
+        dz = reshape(unthunk(z̄), length(y), length(x))
+        x̄ = @thunk conj.(dz' * y)
+        ȳ = @thunk dz * conj.(x)
+        return NoTangent(), x̄, ȳ
+    end
+    return kron(x, y), kron_pullback
+end
+
+function rrule(::typeof(kron), x::AbstractMatrix{<:Number}, y::AbstractVector{<:Number})
+    function kron_pullback(z̄)
+        dz = reshape(unthunk(z̄), length(y), size(x)...)
+        x̄ = @thunk conj.(dot.(eachslice(dz; dims = (2, 3)), Ref(y)))
+        ȳ = @thunk conj.(dot.(eachslice(dz; dims = 1), Ref(x)))
+        return NoTangent(), x̄, ȳ
+    end
+    return kron(x, y), kron_pullback
+end
+
+function rrule(::typeof(kron), x::AbstractVector{<:Number}, y::AbstractMatrix{<:Number})
+    function kron_pullback(z̄)
+        dz = reshape(unthunk(z̄), size(y, 1), length(x), size(y, 2))
+        x̄ = @thunk conj.(dot.(eachslice(dz; dims = 2), Ref(y)))
+        ȳ = @thunk conj.(dot.(eachslice(dz; dims = (1, 3)), Ref(x)))
+        return NoTangent(), x̄, ȳ
+    end
+    return kron(x, y), kron_pullback
+end
+
+function rrule(::typeof(kron), x::AbstractMatrix{<:Number}, y::AbstractMatrix{<:Number})
+    function kron_pullback(z̄)
+        dz = reshape(unthunk(z̄), size(y, 1), size(x, 1), size(y, 2), size(x, 2))
+        x̄ = @thunk conj.(dot.(eachslice(dz, dims = (2, 4)), Ref(y)))
+        ȳ = @thunk conj.(dot.(eachslice(dz; dims = (1, 3)), Ref(x)))
```

Comment:
Here can't you reverse the order of the arguments to `dot` to eliminate the call to `conj`?